finder.ts

```diff
@@ -147,10 +147,13 @@ function findUniquePath(
   stack: Knot[][],
   fallback?: () => Path | null
 ): Path | null {
-  const paths = sort(combinations(stack))
-  if (paths.length > config.threshold) {
-    return fallback ? fallback() : null
+  // Check first the total number of combinations first since generating the combinations can cause memory exhaustion
+  const numCombinations = stack.reduce((acc, i) => acc * i.length, 1);
```

Comment:
I'm not sure about this calculation logic. Better more the if into `combinations()` generator and throw an error of limit.